router/throttler/factory.go

```diff
@@ -0,0 +1,137 @@
+package throttler
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/go-redis/redis/v8"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-go-kit/throttling"
+)
+
+type Factory interface {
+	Get(destName, destID string) Throttler
+	Shutdown()
+}
+
+// NewFactory constructs a new Throttler Factory
+func NewFactory(config *config.Config, stats stats.Stats) (Factory, error) {
```

Comment:
I am returning the interface here to limit the scope of the consumer.
cc @atzoum

Reply:
@atzoum can elaborate, it seems like an anti-pattern to me

Reply:
You are asking me to comment on a debatable topic :)
My view is that there are valid reasons for choosing to return interfaces instead of concrete implementations, e.g.:
- Hiding implementation details from callers (no need to worry about misuse of exported fields in structs)
- Augmenting/decorating behaviour at runtime
- Succinct documentation of the provided API (methods) to callers  

The above doesn't mean that the callers need to re-use the interface that is being returned as is, they can declare their own (scoped) interfaces if needed.